codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSAuthUtils.kt

```diff
@@ -57,24 +60,22 @@ open class SigV4Utils() {
             val auth = ServiceIndex.of(model).getEffectiveAuthSchemes(service.id, operation.id)
             return auth.containsKey(SigV4Trait.ID) && !operation.hasTrait<OptionalAuthTrait>()
         }
+    }
 
-        fun getModeledAuthSchemesSupportedBySDK(
-            ctx: ProtocolGenerator.GenerationContext,
-            writer: SwiftWriter,
-        ): String {
-            val effectiveAuthSchemes = ServiceIndex(ctx.model).getEffectiveAuthSchemes(ctx.service)
-            var authSchemeList = arrayOf<String>()
+    override fun addAdditionalSchemes(writer: SwiftWriter, authSchemeList: Array<String>): Array<String> {
+        val effectiveAuthSchemes = ServiceIndex(ctx.model).getEffectiveAuthSchemes(ctx.service)
 
-            val sdkId = AuthSchemeResolverGenerator.getSdkId(ctx)
-            val servicesUsingSigV4A = arrayOf("S3", "EventBridge", "CloudFrontKeyValueStore")
+        val sdkId = AuthSchemeResolverGenerator.getSdkId(ctx)
+        val servicesUsingSigV4A = arrayOf("S3", "EventBridge", "CloudFrontKeyValueStore")
+        var updatedAuthSchemeList = authSchemeList
 
-            if (effectiveAuthSchemes.contains(SigV4Trait.ID)) {
-                authSchemeList += writer.format("\$N()", AWSSDKHTTPAuthTypes.SigV4AuthScheme)
-            }
-            if (effectiveAuthSchemes.contains(SigV4ATrait.ID) || servicesUsingSigV4A.contains(sdkId)) {
-                authSchemeList += writer.format("\$N()", AWSSDKHTTPAuthTypes.SigV4AAuthScheme)
-            }
-            return "[${authSchemeList.joinToString(", ")}]"
+        if (effectiveAuthSchemes.contains(SigV4Trait.ID)) {
+            updatedAuthSchemeList += writer.format("\$N()", AWSSDKHTTPAuthTypes.SigV4AuthScheme)
+        }
+        if (effectiveAuthSchemes.contains(SigV4ATrait.ID) || servicesUsingSigV4A.contains(sdkId)) {
+            updatedAuthSchemeList += writer.format("\$N()", AWSSDKHTTPAuthTypes.SigV4AAuthScheme)
         }
+
+        return updatedAuthSchemeList
```

Comment:
Handles detecting and adding the SigV4 / SigV4A auth schemes to list of auth schemes supported by the service.